backend/Controllers/user.js

```diff
@@ -0,0 +1,101 @@
+const Address = require("../Database/address");
+const Cart = require("../Database/cart");
+const User = require("../Database/user");
+const Wishlist = require("../Database/wishlist");
+
+
+//Error constructors
+class LoginError extends Error {
+    constructor(message) {
+      super(message);
+      this.name = "LoginError";
+    }
+  }
+
+class AuthError extends Error {
+    constructor(message) {
+      super(message);
+      this.name = "AuthError";
+    }
+  }
+
+//controllers
+    exports.findUserById = async (req, res, next, id) => {
+     await User.findById(id).exec(( err, user) => {
+         if(err) {
+             return res.status(404).json({
+                 success:false, message: "User Not found"
+             })
+         }
+         req.user = user;
+     })
+    }
+
+    exports.getUsersFromDatabase = async (req, res) =>{
+        try{
+            const users = await User.find({});
+            res.json({users: users, success: true})
```

Comment:
Make sure that the password is either not being sent in the response otherwise it is encoded.